extensions/pdh/PerformanceDataMonitor.cpp

```diff
@@ -264,16 +281,29 @@ void PerformanceDataMonitor::setupMembersFromProperties(const std::shared_ptr<co
 
   std::string output_format_string;
   if (context->getProperty(OutputFormatProperty.getName(), output_format_string)) {
-    if (output_format_string == OPEN_TELEMETRY_FORMAT_STR) {
-      logger_->log_trace("OutputFormat is configured to be OpenTelemetry");
+    if (output_format_string == PRETTY_OPEN_TELEMETRY_FORMAT_STR || output_format_string == COMPACT_OPEN_TELEMETRY_FORMAT_STR) {
       output_format_ = OutputFormat::OPENTELEMETRY;
-    } else if (output_format_string == JSON_FORMAT_STR) {
-      logger_->log_trace("OutputFormat is configured to be JSON");
+      pretty_output_ = output_format_string == PRETTY_OPEN_TELEMETRY_FORMAT_STR;
+      logger_->log_trace("OutputFormat is configured to be %s OpenTelemetry", pretty_output_ ? "pretty" : "compact");
+    } else if (output_format_string == PRETTY_JSON_FORMAT_STR || output_format_string == COMPACT_JSON_FORMAT_STR) {
       output_format_ = OutputFormat::JSON;
+      pretty_output_ = output_format_string == PRETTY_JSON_FORMAT_STR;
+      logger_->log_trace("OutputFormat is configured to be %s JSON", pretty_output_ ? "pretty" : "compact");
     } else {
-      logger_->log_error("Invalid OutputFormat, defaulting to JSON");
       output_format_ = OutputFormat::JSON;
+      pretty_output_ = true;
+      logger_->log_error("Invalid OutputFormat, defaulting to %s JSON", pretty_output_ ? "pretty" : "compact");
```

Comment:
we should throw here instead of defaulting, the user gave an invalid output format, that is an error IMO

Reply:
good idea changed it in [26802f7](https://github.com/martinzink/nifi-minifi-cpp/commit/26802f77ef539757fcedbffea0f596b2ef0cc828)

Reply:
although `std::invalid_argument` makes sense, would throwing an `Exception(PROCESS_SCHEDULE_EXCEPTION)` make it more consistent with other processors?

Reply:
changed it in [0844651](https://github.com/martinzink/nifi-minifi-cpp/commit/084465104f3b63c29485638adb1c62d2b3b7e0eb)